frontend/src/core/editor/selectors.ts

```diff
@@ -32,58 +31,51 @@ export function _existingTranslation(
                 typeof initialTranslation !== 'string' &&
                 translation.equals(initialTranslation)))
     ) {
-        existingTranslation = activeTranslation;
-    } else if (history.translations.length) {
-        // If translation is a FluentMessage, from the fluent editor.
-        if (typeof translation !== 'string') {
-            // We apply a bunch of logic on the stored translation, to
-            // make it work with our Editor. So here, we need to
-            // re-serialize it and re-parse it to make sure the translation
-            // object is a "clean" one, as produced by Fluent. Otherwise
-            // we encounter bugs when comparing it with the history items.
-            const fluentTranslation = fluent.parser.parseEntry(
-                fluent.serializer.serializeEntry(translation),
-            );
-            existingTranslation = history.translations.find((t) =>
-                fluentTranslation.equals(fluent.parser.parseEntry(t.string)),
+        return activeTranslation;
+    }
+    if (!history.translations.length) {
+        return;
+    }
```

Comment:
```suggestion
    if (history.translations.length === 0) {
        return undefined;
    }
```

It's better to be explicit about noting that we care about the return value, even if it's `undefined`.